tests/e2e/tests/checkout/checkout.page.ts

```diff
@@ -0,0 +1,258 @@
+/**
+ * External dependencies
+ */
+import { Page } from '@playwright/test';
+import { expect } from '@woocommerce/e2e-playwright-utils';
+
+export class CheckoutPage {
+	private BLOCK_NAME = 'woocommerce/checkout';
+	private page: Page;
+
+	constructor( { page }: { page: Page } ) {
+		this.page = page;
+	}
+
+	async goToCheckout() {
+		await this.page.goto( '/checkout', {
+			waitUntil: 'commit',
+		} );
+	}
+
+	async isShippingRateSelected(
+		shippingName: string,
+		shippingPrice: string
+	) {
+		const shippingLine = this.page.locator(
+			'.wc-block-components-totals-shipping'
+		);
```

Comment:
Do you think we could use [this approach](https://github.com/microsoft/playwright/issues/13286#issuecomment-1088011833) to avoid using CSS selector? It's still a CSS selector, but at least we don't need to use class.

``` 
page.locator('div', { has: page.locator('text="Playwright"') })
```

Reply:
Good suggestion, however I tried this and I couldn't get it to work well due to "shipping" being on the page multiple times, therefore requiring a chain of selectors, in my opinion the code was harder to reason with compared to using just the class name.

Reply:
I combined the tests into a single run, and also brought in the test: `User can have different shipping and billing addresses` to this one. Favouring longer-running tests that test more functionality is better as we reduce setup/teardown time, and it makes sense to test these things together.